### ISSUE TITLE
bug: all imports are not well managed when generating ModelRoleHandlers by CtGenerationTest

### ISSUE THREAD

**Issue by surli** [INRIA/spoon#2076]
The following imports are added but wrong after copy/paste the file:

```
import spoon.generating.meta.spoon.reflect.meta.impl.ListHandler;
import spoon.generating.meta.spoon.reflect.meta.impl.MapHandler;
import spoon.generating.meta.spoon.reflect.meta.impl.SetHandler;
import spoon.generating.meta.spoon.reflect.meta.impl.SingleHandler;
```

**Comment by tdurieux** [comment]
Wrong or unused?

**Comment by surli** [comment]
Wrong.

**Comment by pvojtechovsky** [comment]
Wrong, yes the class name is prefixed with unexpected packages. I will try to debug it

**Comment by pvojtechovsky** [comment]
It is not reproducible on Spoon master. I will try it on your PR. But which one it was?

**Comment by pvojtechovsky** [comment]
There are at least two problems in handling of type references in new Pattern
1) the legacy templates replaced all types whose simple name was equal to template parameter name. This is easy to fix.
2) the legacy templates replaced only some attributes of CtTypeReference. It replaced: simpleName, packageReference, declaringType. But it did not replaced actual type arguments. New Pattern replaces either one attribute or whole node. So I have to improve it here

**Comment by surli** [comment]
It has been fixed in #2078 Discussion continue for point 2 in #2077.
